internal/datasource/config/config_writer.go

```diff
@@ -82,23 +89,29 @@ func (cw *ConfigWriter) Rollback(ctx context.Context, skippedFiles CacheContent,
 	return nil
 }
 
-func (cw *ConfigWriter) Write(ctx context.Context, filesURL,
-	tenantID, instanceID string,
+// has cognitive-complexity of 11 due to the number or nil and error checks
+// nolint: revive
+func (cw *ConfigWriter) Write(ctx context.Context,
+	request *v1.ManagementPlaneRequest_ConfigApplyRequest, instanceID string,
```

Comment:
Can we get the `instanceID` from the `request` parameter?

Reply:
Good spot I meant to do that and forgot